src/Infrastructure/Data/CatalogContext.cs

```diff
@@ -23,134 +22,11 @@ public CatalogContext(DbContextOptions<CatalogContext> options) : base(options)
-                .IsRequired();
-
-            builder.Property(a => a.State)
-                .HasMaxLength(60);
-
-            builder.Property(a => a.Country)
-                .HasMaxLength(90)
-                .IsRequired();
-
-            builder.Property(a => a.City)
-                .HasMaxLength(100)
-                .IsRequired();
-        }
-
-        private void ConfigureBasket(EntityTypeBuilder<Basket> builder)
-        {
-            var navigation = builder.Metadata.FindNavigation(nameof(Basket.Items));
-
-            navigation.SetPropertyAccessMode(PropertyAccessMode.Field);
-        }
-
-        private void ConfigureCatalogItem(EntityTypeBuilder<CatalogItem> builder)
-        {
-            builder.ToTable("Catalog");
-
-            builder.Property(ci => ci.Id)
-                .ForSqlServerUseSequenceHiLo("catalog_hilo")
-                .IsRequired();
-
-            builder.Property(ci => ci.Name)
-                .IsRequired(true)
-                .HasMaxLength(50);
-
-            builder.Property(ci => ci.Price)
-                .IsRequired(true)
-                .HasColumnType("decimal(18,2)");
-
-            builder.Property(ci => ci.PictureUri)
-                .IsRequired(false);
-
-            builder.HasOne(ci => ci.CatalogBrand)
-                .WithMany()
-                .HasForeignKey(ci => ci.CatalogBrandId);
-
-            builder.HasOne(ci => ci.CatalogType)
-                .WithMany()
-                .HasForeignKey(ci => ci.CatalogTypeId);
-        }
-
-        private void ConfigureCatalogBrand(EntityTypeBuilder<CatalogBrand> builder)
-        {
-            builder.ToTable("CatalogBrand");
-
-            builder.HasKey(ci => ci.Id);
-
-            builder.Property(ci => ci.Id)
-               .ForSqlServerUseSequenceHiLo("catalog_brand_hilo")
-               .IsRequired();
-
-            builder.Property(cb => cb.Brand)
-                .IsRequired()
-                .HasMaxLength(100);
-        }
-
-        private void ConfigureCatalogType(EntityTypeBuilder<CatalogType> builder)
-        {
-            builder.ToTable("CatalogType");
-
-            builder.HasKey(ci => ci.Id);
-
-            builder.Property(ci => ci.Id)
-               .ForSqlServerUseSequenceHiLo("catalog_type_hilo")
-               .IsRequired();
-
-            builder.Property(cb => cb.Type)
-                .IsRequired()
-                .HasMaxLength(100);
-        }
-
-        private void ConfigureOrder(EntityTypeBuilder<Order> builder)
-        {
-            var navigation = builder.Metadata.FindNavigation(nameof(Order.OrderItems));
-
-            navigation.SetPropertyAccessMode(PropertyAccessMode.Field);
-
-            builder.OwnsOne(o => o.ShipToAddress);
-        }
-
-        private void ConfigureOrderItem(EntityTypeBuilder<OrderItem> builder)
-        {
-            builder.OwnsOne(i => i.ItemOrdered);
+            base.OnModelCreating(builder);
+            builder.ApplyAllConfigurationsFromCurrentAssembly();
```

Comment:
I really like cleaning this class up and splitting the configurations out in to individual files. It really makes reading and understanding what is going on less jarring. Good change.